### ISSUE TITLE
Invalid error:  *[alert] ignoring stale global SSL .... while SSL handshaking to upstream* appears in kong logs when using jwt plugin with RSA in 2.0.1

### ISSUE THREAD

**Issue by ealogar** [Kong/kong#5614]
### Summary

Using the jwt plugin with RSA keys with kong version 2.0.1 gives this alert in kong logs:
```
2020/02/27 00:40:30 [alert] 22#0: *358 ignoring stale global SSL error (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: ANY PRIVATE KEY error:0909006C:PEM routines:get_name:no start line:Expecting: ANY PRIVATE KEY) while SSL handshaking to upstream, client: 172.18.0.1, server: kong, request: "GET /?
```
There is not any failure and the plugin is working well; it authenticates the consumer and forward the request to the upstream, but this error appears in logs and it shouldn't.

### Steps To Reproduce

1. Follow the steps in https://docs.konghq.com/install/docker/ to run a kong version in DB-less mode
2. Use this declarative config:
```
_format_version: "1.1"

services:
- name: my-service
  url: https://example.com
  plugins:
  - name: jwt
    config:
  routes:
  - name: my-route
    paths:
    - /

consumers:
- username: my-user
  custom_id: my-user

jwt_secrets:
- consumer: my-user
  algorithm: RS256
  key: my-issuer
  rsa_public_key: |
    -----BEGIN PUBLIC KEY-----
    MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAoltAOa+9RQY7QpHoXzNO
    6oAN53yZZnHN6XDmg4Rmm9sIPbxyWLG/KvctxvdXgJPjdXw4HsIS9g6yb0k3Hiji
    whf0E7C9pRUEUY+7A2AaQkLYx5KihWzFgyhqkZu0+MfhLVbe4GiAHiShT0ELhg22
    MDO5nVgOTvNnEzDQWBqFQQ8Bxxco+htTcl35WkTaeoNQsefKF5A8jzESpmmC4D5e
    IXeCOcgF+g+vWXXz9kgZD9x0hZdBtUKO/oRY7AkGnWxL8RBH29DcVJ9cjGTw4YeL
    PiCuaa7WoYeE7lW3JTXdvLLZzA5HVtCZdUVo+7dgrBwI2bYYCZWGzIymUaRvZSa3
    wwIDAQAB
    -----END PUBLIC KEY-----

```

I have used the private.key and public.key provided in spec file of jwt tests, but you can use whatever

3. Go to jwt.io and generate a JWT following https://docs.konghq.com/hub/kong-inc/jwt/#craft-a-jwt-with-publicprivate-keys-rs256-or-es256
Basically :
Define this header
```
{
    "typ": "JWT",
    "alg": "RS256"
}
```
and a payload like this:
```
{
     "iss": "my-issuer"
}
```

include the private key and you will have valid jwt to call the api:

4. Do the call to the api: 
```
curl http://localhost:8000/?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJteS1pc3N1ZXIifQ.Zq53VtPi1Ck6ygsAbXTsrLcJl0bi0-YwdxDxtE8OmwlO0DsqXrUlOZlI_g8tV_OulEnfyY-seJ2fFvIoIv2J5P2DINXzgYGvzZQ5f1_MyVZCQIKyiAD1O0qCARQsBmz-Egp74ejwRck139CaH6GDuL2Fn8DP7XmNL9tjOL9EQ8Z4-VLfPSeDhQ4XeOoRJeCDrvdCVbzyzFclDE1ZkHXVZZDOI9Ga31GZyxbDLl0Pv7OHhIMe3WsZfu2yf7C0HBgKIvJ6VFZnYdTHQ0O2N2FmLPwbzv4gtRJ-viYojEl76qp8Y8JmnNIYv946DPEh2yglKHj5dH8pHT5RjI_uHUD2FA
```

You are forwarded correctly to example.com but the alert error message appears in the logs and it's annoying

### Additional Details & Logs

I have checked that this is produced since the luaossl was changed in the plugin by lua-resty-openssl.
Every time you do a pkey:verify in the plugin (see line 45 of jwt_parser.lua) this error appears.
It sound to me as the public pem that you use to verify the token appears somehow in the ssl handshake ... and the error says that it's not a private key (of course because it should not be...)
Notice that I haven't configured any two way ssl verification or similar.

If i changed the jwt_parser to use luaossl back the error dissapears (I have also checked that the library is still provided in kong 2.0.1).

- Kong version 2.0.1
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs 
```
2020/02/27 00:59:08 [alert] 22#0: *9310 ignoring stale global SSL error (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: ANY PRIVATE KEY) while SSL handshaking to upstream, client: 172.18.0.1, server: kong, request: "GET /?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJteS1pc3N1ZXIifQ.Zq53VtPi1Ck6ygsAbXTsrLcJl0bi0-YwdxDxtE8OmwlO0DsqXrUlOZlI_g8tV_OulEnfyY-seJ2fFvIoIv2J5P2DINXzgYGvzZQ5f1_MyVZCQIKyiAD1O0qCARQsBmz-Egp74ejwRck139CaH6GDuL2Fn8DP7XmNL9tjOL9EQ8Z4-VLfPSeDhQ4XeOoRJeCDrvdCVbzyzFclDE1ZkHXVZZDOI9Ga31GZyxbDLl0Pv7OHhIMe3WsZfu2yf7C0HBgKIvJ6VFZnYdTHQ0O2N2FmLPwbzv4gtRJ-viYojEl76qp8Y8JmnNIYv946DPEh2yglKHj5dH8pHT5RjI_uHUD2FA HTTP/1.1", upstream: "https://93.184.216.34:443/?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJteS1pc3N1ZXIifQ.Zq53VtPi1Ck6ygsAbXTsrLcJl0bi0-YwdxDxtE8OmwlO0DsqXrUlOZlI_g8tV_OulEnfyY-seJ2fFvIoIv2J5P2DINXzgYGvzZQ5f1_MyVZCQIKyiAD1O0qCARQsBmz-Egp74ejwRck139CaH6GDuL2Fn8DP7XmNL9tjOL9EQ8Z4-VLfPSeDhQ4XeOoRJeCDrvdCVbzyzFclDE1ZkHXVZZDOI9Ga31GZyxbDLl0Pv7OHhIMe3WsZfu2yf7C0HBgKIvJ6VFZnYdTHQ0O2N2FmLPwbzv4gtRJ-viYojEl76qp8Y8JmnNIYv946DPEh2yglKHj5dH8pHT5RjI_uHUD2FA", host: "localhost:8000"
172.18.0.1 - - [27/Feb/2020:00:59:09 +0000] "GET /?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJteS1pc3N1ZXIifQ.Zq53VtPi1Ck6ygsAbXTsrLcJl0bi0-YwdxDxtE8OmwlO0DsqXrUlOZlI_g8tV_OulEnfyY-seJ2fFvIoIv2J5P2DINXzgYGvzZQ5f1_MyVZCQIKyiAD1O0qCARQsBmz-Egp74ejwRck139CaH6GDuL2Fn8DP7XmNL9tjOL9EQ8Z4-VLfPSeDhQ4XeOoRJeCDrvdCVbzyzFclDE1ZkHXVZZDOI9Ga31GZyxbDLl0Pv7OHhIMe3WsZfu2yf7C0HBgKIvJ6VFZnYdTHQ0O2N2FmLPwbzv4gtRJ-viYojEl76qp8Y8JmnNIYv946DPEh2yglKHj5dH8pHT5RjI_uHUD2FA HTTP/1.1" 200 1256 "-" "curl/7.63.0"
```

**Comment by bungle** [comment]
Ping @fffonion, is this by the `lua-resty-openssl`?

**Comment by fffonion** [comment]
Yes this is indeed an issue with lua-resty-openssl prior to 0.5.0 when pkey is loaded with implict type, the error is not properly cleared. There's no side effect other than a error log.

It will be fixed in kong 2.1. I'll bump the version for 2.0 as well.

**Comment by ealogar** [comment]
I can confirm that 0.4.4 fix the issue!
